BJS2-3491 - Система постов - Unicorn BC3

hhzktoeto opened the pull request.

hhzktoeto committed:
Добавил основные методы + поправки в клиентах

- добавил возможность создать, получить, опубликовать, обновить,  удалить пост
- в ProjectServiceClient и UserServiceClient исправил эндпоинты, в соответствии с тем, как они написаны в самих сервисах

hhzktoeto committed:
Вся логика без тестов

hhzktoeto committed:
Написал тесты на PostService

hhzktoeto committed:
Написал тесты на PostValidator

hhzktoeto committed:
Мелкие поправки

snailrake commented on src/main/java/faang/school/postservice/service/post/PostService.java:
```diff
@@ -0,0 +1,102 @@
+        }
+        if (projectId != null) {
+            postValidator.validateIfProjectExistsById(projectId);
+        }
```
Почему бы это просто в валидаторе не оставить?

snailrake commented on src/test/java/faang/school/postservice/service/post/PostServiceTest.java:
```diff
@@ -0,0 +1,257 @@
+                () -> verify(postRepository, times(1)).save(firstPost),
+                () -> verify(postMapper, times(1)).toEntity(firstPostDto),
+                () -> verify(postMapper, times(1)).toDto(firstPost),
+                () -> assertEquals(returned, firstPostDto)
```
В assertEquals первый аргумент - ожидаемый, второй - фактический. В остальных так же

hhzktoeto committed:
Исправления после cross-review

hhzktoeto committed:
Добавил обработку FeignException в PostValidator

hhzktoeto committed:
Дописал тесты, добавил Builder в DTOs

hhzktoeto committed:
Добавил валидацию

добавил проверку на то, является ли пост удалённым (перед удалением или публикацией)

IrinaVereshchagina13 commented on src/main/java/faang/school/postservice/controller/post/PostController.java:
```diff
@@ -0,0 +1,69 @@
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
```
Давай добавим аннотации сваггера

hhzktoeto replied:
В следующей задаче с альбомами (3514) я добавил аннотации. 
Просто когда я писал эту задачу, я ещё не имплементировал сваггер в проект

По задаче с альбомами осталось дописать пару тестов и я отправлю на ревью
